rio_tiler/reader.py

```diff
@@ -580,7 +582,15 @@ def point(
         ):
             raise PointOutsideBounds("Point is outside dataset bounds")
 
-        row, col = dataset.index(lon, lat)
+        if interpolate:
+            # Ref: https://github.com/cogeotiff/rio-tiler/issues/793
+            # https://github.com/OSGeo/gdal/blob/a3d68b069e6b3676ba437faca5dca6ae2076ce24/swig/python/gdal-utils/osgeo_utils/samples/gdallocationinfo.py#L185-L197
+            rows, cols = rowcol(dataset.transform, xs=[lon], ys=[lat], op=lambda x: x)
```

Comment:
we use `op=lambda x: x` to make sure rasterio doesn't apply any `rounding` on the pixel coordinates

FYI we can't use `dataset.index(lon, lat, op=lambda x: x)` because rasterio will use `int()` https://github.com/rasterio/rasterio/blob/main/rasterio/transform.py#L135 😓